core/kt-osrd-utils/src/main/kotlin/fr/sncf/osrd/utils/AppendOnlySet.kt

```diff
@@ -25,12 +27,18 @@ class AppendOnlySet<T>(
      * end. Pre-filtered using a bloom filter.
      */
     fun contains(key: T): Boolean {
-        if (!keyFilter.mayContain(key)) return false
+        if (!keyFilter.mightContain(key)) return false
         return list.findLast { it == key } != null
     }
 }
 
 /** Returns a new empty set */
-fun <T> appendOnlySetOf(): AppendOnlySet<T> {
-    return AppendOnlySet(appendOnlyLinkedListOf(), emptyBloomFilter())
+inline fun <reified T> appendOnlySetOf(
+    expectedInsertions: Int = 1500,
+    falsePositiveRate: Double = 0.03,
+): AppendOnlySet<T> {
```

Comment:
The naming is unfortunate. If `appendOnlySetOf` accept arguments, one would expect they are the elements of the created set, not that it always return an empty set.

Maybe this should just be a secondary constructor of AppendOnlySet instead?

Reply:
Good point, I'll rename the method